smurfs/src/reducers/index.js

```diff
@@ -21,3 +26,45 @@
   There is no need for 'combineReducers' in this project.
   Components can then read your store as, `state` and not `state.fooReducer`.
 */
+function reducer(state = initialState, action){
+  console.log("reducer", action)
+  switch(action.type){
+    case FETCH_START:
+      return {
+        ...state, 
+        error: "",
+        smurfs: [], 
+        fetchingSmurfs: true
+      };
+    case FETCH_SUCCESS:
+      return{
+        ...state, 
+        error: "",
+        smurfs: action.smurfs, 
+        fetchingSmurfs: false
+      };
+    case FETCH_ERROR:
+      return{
+        ...state, 
+        error: "", 
+        fetchingSmurfs: false
+      }
+    case ADD_SMURF:
+    return{
+      ...state, 
+      error: '', 
+      addingSmurf: true
+    }
+    case POST_SUCCESS:
+        return{
+          ...state, 
+          error: '', 
+          addingSmurf: false, 
+          smurfs: [...state.smurfs, action.smurfs]
+        }
+    default: 
+    return state
+  }
+}
+
```

Comment:
You should have a case for when the POST fails, this action type should be coming from the action creator but you should have a case in your switch statement that takes care of this situation.